match/tensor.py

```diff
@@ -65,32 +62,32 @@ def _gradient() -> None:
         result._gradient = _gradient
         return result
 
-    def sum(self) -> Matrix:
+    def sum(self) -> Tensor:
         """Return the sum of all values across both dimensions."""
-        result = Matrix(List2D(1, 1, self.data.sum()), children=(self,))
+        result = Tensor(TensorData(value = self.data.sum()), children=(self,))
 
         def _gradient() -> None:
             info(f"Gradient of summation. Shape: {self.shape}")
-            self.grad += List2D(self.nrow, self.ncol, result.grad.vals[0][0])
+            self.grad += TensorData(*self.shape, value=result.data.item())
 
         result._gradient = _gradient
         return result
 
-    def mean(self) -> Matrix:
+    def mean(self) -> Tensor:
         """Return the mean of all values across both dimensions."""
-        result = Matrix(List2D(1, 1, self.data.mean()), children=(self,))
+        result = Tensor(TensorData(value = self.data.mean()), children=(self,))
 
         def _gradient() -> None:
             info(f"Gradient of mean. Shape: {self.shape}")
-            n = self.nrow * self.ncol
-            self.grad += List2D(self.nrow, self.ncol, result.grad.vals[0][0] / n)
+            n = len(self.data._data) # Equivalent to prod(self.shape), just cheaper.
```

Comment:
Should we have a `num_elements` method like `torch`?

Reply:
Ack. Yes.